### PR TITLE
Reworked pipelines sample app

### PR DESCRIPTION
**Pull request by aL3891**
Split out all the samples into different classes, also added a menu for launching samples. also made base classes for the different sample types
There are some issues with the client samples but because this change is already big i figured i fix that later

### CODE REVIEW

**Comment by aL3891** [comment]
what say you @davidfowl ?
Strange i rebased and resolved that conflict but its not showing up, perhaps i did something wrong?

File: samples/System.IO.Pipelines.Samples/System.IO.Pipelines.Samples.csproj [hunk 1]
```diff
@@ -1,9 +1,13 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <Import Project="..\..\tools\common.props" />
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
-    <OutputType>Exe</OutputType>
+    <OutputType>exe</OutputType>
```
Comment (davidfowl) [inline samples/System.IO.Pipelines.Samples/System.IO.Pipelines.Samples.csproj]: @aL3891 Can you undo this change?

Comment (aL3891) [inline samples/System.IO.Pipelines.Samples/System.IO.Pipelines.Samples.csproj]: Done!

**Comment by ahsonkhan** [comment]
@davidfowl, are you signing off on this PR?

**Comment by davidfowl** [comment]
There's one issue that hasn't been addressed. The csproj updates need to be undone.

**Comment by aL3891** [comment]
but I did fix that @davidfowl ? or was it something else?

**Comment by davidfowl** [comment]
@aL3891 revert the entire csproj change.

**Comment by aL3891** [comment]
oh, ok, i'll rebase

**Comment by aL3891** [comment]
got it

**Comment by aL3891** [comment]
fixed conflicts

**Comment by ahsonkhan** [comment]
@davidfowl, is this good to go?

**Comment by aL3891** [comment]
ok, build is over and done with, hows about this one? :)

**Comment by ahsonkhan** [comment]
Hey @aL3891, @davidfowl. since this PR, I am getting warnings in the build as follows:

58>Framing\Codec.cs(16,27,16,30): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
58>Socket\RawSocketHttpServerSample.cs(18,39,18,44): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
58>Socket\RawSocketHttpClientSample.cs(23,56,23,69): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
58>Rio\RawRioHttpClientSample.cs(23,56,23,69): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
58>SampleBase\AspNetSampleBase.cs(19,27,19,30): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.

Do we ignore these warnings, suppress them, or fix the code to resolve the warnings?

**Comment by aL3891** [comment]
They should most likely be fixed, i can take a look and open another pr
